test/create_game_section/Lobby/lobby_widget_test.dart

```diff
@@ -7,16 +7,64 @@ import 'package:flutter_test/flutter_test.dart';
 import 'package:niira/main.dart';
 import 'package:niira/models/game.dart';
 import 'package:niira/models/user_data.dart';
+import 'package:niira/screens/lobby/lobby.dart';
+import 'package:niira/screens/lobby/list_of_created_games.dart';
+import 'package:niira/services/auth/auth_service.dart';
 import 'package:niira/navigation/navigation.dart';
+import 'package:niira/services/database/database_service.dart';
+import 'package:niira/services/game_service.dart';
+import 'package:niira/services/location_service.dart';
+import 'package:provider/provider.dart';
+
 import '../../mocks/mock_user_data.dart';
 import '../../mocks/services/mock_auth_service.dart';
 import '../../mocks/services/mock_database_service.dart';
+import '../../mocks/services/mock_game_service.dart';
+import '../../mocks/services/mock_location_service.dart';
 import '../../mocks/services/mock_firebase_platform.dart';
 
 void main() {
   setUp(() {
     Firebase.delegatePackingProperty = MockFirebasePlatform();
   });
+  testWidgets(
+      'show loading icon until have users location, then show list of created games ',
+      (WidgetTester tester) async {
+    // init services
+    final controller = StreamController<UserData>();
+    final navigation = Navigation();
+    final mockAuthService = MockAuthService(controller: controller);
+    final mockLocationService = MockLocationService();
+    final mockGameService = FakeGameService();
+    final mockDatabseController = StreamController<List<Game>>();
+    final mockDatabaseService =
+        MockDatabaseService(controller: mockDatabseController);
+
+    // create the widget under test
+    await tester.pumpWidget(
+      MultiProvider(
+        providers: [
+          Provider<AuthService>.value(value: mockAuthService),
+          Provider<Navigation>.value(value: navigation),
+          Provider<LocationService>.value(value: mockLocationService),
+          Provider<DatabaseService>.value(value: mockDatabaseService),
+          Provider<GameService>.value(value: mockGameService),
+        ],
+        child: MaterialApp(
+          home: LobbyScreen(),
+        ),
+      ),
+    );
+
+    // show loading icon
+    expect(find.byKey(Key('loading_indicator')), findsOneWidget);
+
+    // get users location
+    await tester.pumpAndSettle();
+
+    // show list of created games
+    expect(find.byType(ListOfCreatedGames), findsOneWidget);
+  });
   testWidgets(
       'shows loading icon + redirects to welcome screen on successfull logout',
       (WidgetTester tester) async {
```

Comment:
all added code under this comment was to fix existing test.
(sorry if im just pointing out the obvious)